docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,493 @@
+
+
+Auto Schema Registration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, client applications automatically register new schemas.
+If they produce new messages to a new topic, then they will automatically try to register new schemas.
+This is very convenient in development environments, but in production environments we recommend that client applications do not automatically register new schemas.
+Users should register schemas outside of the client application to control when schemas are registered with |sr-long| and how they evolve.
+
+Within the application, disable automatic schema registration by setting the configuration parameter `auto.register.schemas=false`, as shown in the examples below.
+
+.. sourcecode:: java
+
+   props.put(AbstractKafkaAvroSerDeConfig.AUTO_REGISTER_SCHEMAS, false);
+
+To manually register the schema outside of the application, send the schema to |sr| and associate it with a subject, in this case `transactions-value`.  It returns a schema id of `1`.
+
+.. sourcecode:: bash
+
+   $ curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" --data '{"schema": "{\"type\":\"record\",\"name\":\"Payment\",\"namespace\":\"io.confluent.examples.clients.basicavro\",\"fields\":[{\"name\":\"id\",\"type\":\"string\"},{\"name\":\"amount\",\"type\":\"double\"}]}"}' http://localhost:8081/subjects/transactions-value/versions
+   {"id":1}
+
+
+Schema Evolution and Compatibility
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Changing Schemas
+^^^^^^^^^^^^^^^^
+
+You have seen the benefit of |sr-long| as being centralized schema management that enables client applications to register and retrieve globally unique schema ids.
+The main value, however, is in enabling schema evolution.
+Similar to how APIs evolve and need to be compatible for all applications that rely on old and new versions of the API, schemas also evolve and likewise need to be compatible for all applications that rely on old and new versions of the schema.
+This schema evolution is a natural behavior of how applications and data develop over time.
+
+|sr-long| allows for schema evolution and provides compatibility checks to ensure that the contract between producers and consumers is not broken.
+This is especially important in Kafka because producers and consumers are decoupled applications that are sometimes developed by different teams.
+Compatibility checks on schemas allow producers and consumers to update independently and evolve their schemas independently, with assurances that they can read new and legacy data.
+
+These are the types of `compatibility types <https://docs.confluent.io/current/avro.html#data-serialization-and-evolution>`_:
+
+* ``FORWARD``: consumers using older schemas can read data written by producers using newer schemas
+* ``BACKWARD``: consumers using newer schemas can read data written by producers using older schemas
+* ``FULL``: forward and backward compatible
+* ``NONE``: schema compatibility checks disabled
```

Comment:
Should the transitive compatibility levels be mentioned? 

https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroCompatibilityLevel.java#L50-L55

Reply:
Ack.